lib/resource-enrichers/knative-service-enricher.js

```diff
@@ -0,0 +1,90 @@
+/*
+ *
+ *  Copyright 2016-2017 Red Hat, Inc, and individual contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+'use strict';
+
+const _ = require('lodash');
+const objectMetadata = require('../definitions/object-metadata');
+
+const baseServiceConfig = {
+  apiVersion: 'serving.knative.dev/v1',
+  kind: 'Service',
+  spec: {} // Should be populated in the resource coming in
+};
+
+// Returns a default knative serving service.
+function defaultService (config) {
+  const serviceConfig = _.merge({}, baseServiceConfig);
+
+  // Apply MetaData
+  serviceConfig.metadata = objectMetadata({
+    name: config.projectName,
+    namespace: config.namespace.name
+  });
+
+  // TODO(lholmquist): Should this also take into account docker images that are not in the openshift registry
```

Comment:
Good question... probably. But I guess that means adding a `--registry` flag or something. So many flags it's like the UN! :rofl:

Reply:
:facepalm:

Reply:
Maybe this is something we do in the future,  but probably not for this PR